frontend/src/scenes/session-recordings/player/snapshot-processing/process-all-snapshots.test.ts

```diff
@@ -272,4 +312,341 @@ describe('process all snapshots', () => {
+                data: [
+                    {
+                        type: 3,
+                        timestamp: 1000,
+                        data: {
+                            source: 0,
+                            adds: [],
+                            removes: [],
+                            texts: [],
+                            attributes: [],
+                        },
+                    },
+                ],
+            })
+
+            const result = await parseEncodedSnapshots([snapshotJson], sessionId)
+
+            expect(result).toHaveLength(1)
+            expect(result[0].windowId).toBe('1')
+            expect(result[0].type).toBe(3)
+
+            const hasFullSnapshot = result.some((r) => r.type === 2)
+            expect(hasFullSnapshot).toBe(false)
+        })
+
+        it('creates synthetic snapshot with correct windowId from original event', async () => {
+            const sessionId = 'test-mobile-session'
+
+            const snapshotJson = JSON.stringify({
+                window_id: 'custom-window-123',
+                data: [
+                    {
+                        type: 3,
+                        timestamp: 2000,
+                        data: {
+                            source: 0,
+                            updates: [
+                                {
+                                    wireframe: {
+                                        type: 'screenshot',
+                                        base64: 'data:image/webp;base64,test',
+                                    },
+                                },
+                            ],
+                        },
+                    },
+                ],
+            })
+
+            const result = await parseEncodedSnapshots([snapshotJson], sessionId)
+
+            result.forEach((event) => {
+                expect(event.windowId).toBe('custom-window-123')
+            })
+
+            const fullSnapshot = result.find((r) => r.type === 2)
+            expect(fullSnapshot?.timestamp).toBe(1999)
+        })
+
+        it('handles multiple mobile events correctly (only first gets synthetic)', async () => {
+            const sessionId = 'test-mobile-session'
+
+            const snapshotJson = JSON.stringify({
+                window_id: '1',
+                data: [
+                    {
+                        type: 3,
+                        timestamp: 1000,
+                        data: {
+                            source: 0,
+                            updates: [{ wireframe: { type: 'screenshot' } }],
+                        },
+                    },
+                    {
+                        type: 3,
+                        timestamp: 2000,
+                        data: {
+                            source: 0,
+                            updates: [{ wireframe: { type: 'screenshot' } }],
+                        },
+                    },
+                ],
+            })
```

Comment:
nit picking, i'd be tempted to pull these json out to a `makeSnapshot` function
will be easier to see what (if anything) changes between tests too then

there's a lot to read otherwise :)